bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -283,30 +283,43 @@ public boolean accept(long journalId) {
         WriteCallback cb;
         Object ctx;
         long enqueueTime;
+        boolean ackBeforeForce;
```

Comment:
I would suggest defer adding these two variables until you really need them. it isn't clear how callback will be executed and the object will be recycled in some branches. this pull request should just take the flags and acknowledge before sync.

Reply:
I think this is needed, because the lifecycle of the QueueEntry is different, we can't simply fire the callback and forget about the entry, see my comments below